File: src/sports_management/modules/notification_module/lib/domain/usecases/get_notifications_usecase.dart
import 'package:server_module/server_module.dart';
import '../repositories/notification_repository.dart';

class GetNotificationsUseCase {
  const GetNotificationsUseCase(this._repository);

  final AppNotificationRepository _repository;

  Future<List<NotificationEntity>> call() {
    return _repository.getNotifications();
  }
}

File: src/react-staff-admin/src/features/notification/domain/usecases/get_notifications.usecase.ts
import { NotificationRepository } from '../repositories/notification.repository';
import { Notification } from '../entities/notification.entity';

export class GetNotificationsUseCase {
  constructor(private repository: NotificationRepository) {}

  async execute(): Promise<{ items: Notification[]; unreadCount: number }> {
    return this.repository.getNotifications();
  }
}
